src/js/angular/controllers.js

```diff
@@ -716,6 +719,40 @@ function mainCtrl($scope, $menuItems, $jwtAuth, $http, toastr, $location, $repos
         }
     }
 
+    /**
+     * Creates a debounced function that delays invoking the provided function until after a specified wait time has
+     * elapsed since the last time the debounced function was invoked. Optionally, the function can be invoked immediately
+     * and then wait for the specified time before allowing the next invocation.
+     * @param {Function} func - The function to debounce.
+     * @param {number} wait - The number of milliseconds to delay.
+     * @param {boolean} [callImmediately=false] - If true, the function will be invoked immediately, then wait for the specified time
+     *                                           before allowing the next invocation.
+     * @return {Function} - Returns the new debounced function.
+     */
+    function debounce(func, wait, callImmediately) {
```

Comment:
Two notes here. First, why this is here and not in some utility and second, why not just using the lodash's debounce function?

Reply:
The lodash debounce doesn't always work so well in this case. The labels still sometimes load slower. I will move this to a utility class instead.

Reply:
Well, how that can be?! :) Did you try tweaking the debounce parameters? Try passing `{trailing: false}`.